### PR TITLE
Create a new RAO: LinearRangeActionRao

### PR DESCRIPTION
**Pull request by TheMaskedTurtle**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Structure of a new remedial action optimizer that deals only with linear remedial actions

### CODE REVIEW

File: ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRaoParameters.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        return "LinearRangeActionRaoParameters";
+    }
+
+    public SensitivityComputationParameters getSensitivityComputationParameters() {
+        return sensitivityComputationParameters;
+    }
+
+    public LinearRangeActionRaoParameters setSensitivityComputationParameters(SensitivityComputationParameters sensiParameters) {
+        this.sensitivityComputationParameters = Objects.requireNonNull(sensiParameters);
+        return this;
```
Comment (alexandrem-rte) [inline ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRaoParameters.java]: why don't you do a classical setter returning void?

File: ra-optimisation/linear-range-action-rao/pom.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    <parent>
+        <artifactId>farao-ra-optimisation</artifactId>
+        <groupId>com.farao-community.farao</groupId>
+        <version>1.3.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>farao-linear-range-action-rao</artifactId>
+    <packaging>jar</packaging>
+    <name>Modular RAO on linear actions implementation</name>
```
Comment (baptsegu) [inline ra-optimisation/linear-range-action-rao/pom.xml]: I would say that the optimisation is linear, but not the actions. We approximate their impact with linear sensitivity coefficients, yet they are _a priori_ not linear.
Maybe : 
`Modular linear RAO implementation`

File: ra-optimisation/pom.xml [hunk 3]
```diff
@@ -22,6 +22,7 @@
         <module>ra-optimisation-afs-local</module>
         <module>rao-api</module>
         <module>search-tree-rao</module>
+        <module>linear-range-action-rao</module>
```
Comment (baptsegu) [inline ra-optimisation/pom.xml]: alphabetical order

File: ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRaoParameters.java (hunk 1 above)
Comment (TheMaskedTurtle) [reply ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRaoParameters.java]: As it is copied from PowSyBl we'll wait to know if there is a reason

Comment (TheMaskedTurtle) [reply ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRaoParameters.java]: Internet says that this pattern could be useful if you have a lot of parameters to set and you don't want to pass too many arguments in the constructor. For us I think we can remove it by now.

Comment (TheMaskedTurtle) [reply ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRaoParameters.java]: After discussion with Sylvain, we will keep it for parameters classes consistency.

File: ra-optimisation/linear-range-action-rao/pom.xml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    <artifactId>farao-linear-range-action-rao</artifactId>
+    <packaging>jar</packaging>
+    <name>Modular linear RAO implementation</name>
+    <description>Implementation of a RAO that will optimize only linear range actions</description>
+
+    <dependencies>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>farao-rao-api</artifactId>
```
Comment (pengbo-ruidian) [inline ra-optimisation/linear-range-action-rao/pom.xml]: dependency order

Comment (TheMaskedTurtle) [reply ra-optimisation/linear-range-action-rao/pom.xml]: ok

**Pull request merged**